### ISSUE TITLE
inspect not show Mounts info like docker-cli

### ISSUE THREAD

**Issue by jonyhy96** [containerd/nerdctl#483]
will nerdctl support show mounts info just like docker-cli? or if there is any alternative tool to do so?

**Comment by jonyhy96** [comment]
from nerdctl
```bash
root@kind-worker2:~# nerdctl -n k8s.io container inspect 42a02c24a458
[
    {
        "Id": "42a02c24a4588bc9bd909e5df956f480e21e246e0fe6ffa9ad1b03653e339706",
        "Created": "2021-10-30T07:58:26.156628094Z",
        "Path": "/docker-entrypoint.sh",
        "Args": [
            "nginx",
            "-g",
            "daemon off;"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Pid": 702,
            "ExitCode": 0,
            "FinishedAt": "0001-01-01T00:00:00Z"
        },
        "Image": "docker.io/library/nginx:stable-alpine",
        "ResolvConfPath": "",
        "HostnamePath": "",
        "LogPath": "",
        "Name": "",
        "Driver": "overlayfs",
        "Platform": "linux",
        "AppArmorProfile": "",
        "NetworkSettings": {
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "10.244.2.2",
            "IPPrefixLen": 24,
            "MacAddress": "b6:97:9c:3a:7f:6b",
            "Networks": {
                "unknown-eth0": {
                    "IPAddress": "10.244.2.2",
                    "IPPrefixLen": 24,
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "b6:97:9c:3a:7f:6b"
                }
            }
        }
    }
]
```

from docker-cli
```bash
➜  k8simages docker inspect 489eff3d5f02
[
    {
        "Id": "489eff3d5f024449940911230fbf8b080c53efe830b17e3485eadf2a445b5561",
        "Created": "2021-10-29T07:02:06.873184186Z",
        "Path": "/usr/local/bin/entrypoint",
        "Args": [
            "/sbin/init"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 2146,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2021-10-30T07:57:55.981800869Z",
            "FinishedAt": "2021-10-30T07:57:53.734728782Z"
        },
        "Image": "sha256:511faf5b722aafd10fbd9762870bf9a5c0dbf58c7f754b165a88e06136e96acf",
        "ResolvConfPath": "/var/lib/docker/containers/489eff3d5f024449940911230fbf8b080c53efe830b17e3485eadf2a445b5561/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/489eff3d5f024449940911230fbf8b080c53efe830b17e3485eadf2a445b5561/hostname",
        "HostsPath": "/var/lib/docker/containers/489eff3d5f024449940911230fbf8b080c53efe830b17e3485eadf2a445b5561/hosts",
        "LogPath": "/var/lib/docker/containers/489eff3d5f024449940911230fbf8b080c53efe830b17e3485eadf2a445b5561/489eff3d5f024449940911230fbf8b080c53efe830b17e3485eadf2a445b5561-json.log",
        "Name": "/kind-worker2",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": [
            "e461ff315819464ded9b4ced83efbd4f5e11c6dbe5d9bddc87506b3803fb4d09",
            "7b677e5e20f30a18609ae5bf0fe6e96cd9d4802557598729e4b2887d288bc0c6"
        ],
        "HostConfig": {
            "Binds": [
                "/lib/modules:/lib/modules:ro"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "kind",
            "PortBindings": {},
            "RestartPolicy": {
                "Name": "on-failure",
                "MaximumRetryCount": 1
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "CgroupnsMode": "host",
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": true,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": [
                "seccomp=unconfined",
                "apparmor=unconfined",
                "label=disable"
            ],
            "Tmpfs": {
                "/run": "",
                "/tmp": ""
            },
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": null,
            "ReadonlyPaths": null,
            "Init": false
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/26eff50ed5b9b32c0c3f03ab4e8d22700732d3b7a355fe976cdda0f998ddec3a-init/diff:/var/lib/docker/overlay2/4c04bdc4bd9e8e43011959d8d7632a9ac894cbf020dbf1e9f86504d3d586c84b/diff:/var/lib/docker/overlay2/3f7abb82ac28030d5d72990e0199d32f2f6fcf20dbe8f7d23abf288bb2bb3a42/diff:/var/lib/docker/overlay2/325aa5f11ceb939dc7a2fc263efb86c3ec46594b4aebc1e4446ce6ed5ee50b99/diff:/var/lib/docker/overlay2/d0da5c264df045e5265dca529b8af6ac72a8f28ab95e52427c58dec1ea450961/diff:/var/lib/docker/overlay2/4ebee0efb47225ffcd2469718eb1584820448f3abb83b803f13a62f54c47b319/diff:/var/lib/docker/overlay2/c60bcb0b136585cf5f8950ccc3b00320896b21fc198f46f86e64801b2779540c/diff:/var/lib/docker/overlay2/4e908879514d32bef6d4c07bf99f4704816642d0aed5eb72f8757ec5ec38e0a0/diff:/var/lib/docker/overlay2/127e3f8c6259608d8dc2db5f9ee237020851a79470ad408304bd8de6f1dc6206/diff:/var/lib/docker/overlay2/108d0c8fe80c088bdf38346a266929114a83d9948f3f8a4d6d41202424bb489c/diff:/var/lib/docker/overlay2/af1a21c4320f8079aaf514d2c07f8d74fd5f972df15f87822bfccbe8b3fb1ef5/diff:/var/lib/docker/overlay2/e12b6111a992ea28efe43d1079fd85a9daf7130eaaf8dca2c7aa06da3f4c3960/diff:/var/lib/docker/overlay2/1d061929b50da75a77d5ec97fe270657273f98977804cc1a6212090510c0865b/diff",
                "MergedDir": "/var/lib/docker/overlay2/26eff50ed5b9b32c0c3f03ab4e8d22700732d3b7a355fe976cdda0f998ddec3a/merged",
                "UpperDir": "/var/lib/docker/overlay2/26eff50ed5b9b32c0c3f03ab4e8d22700732d3b7a355fe976cdda0f998ddec3a/diff",
                "WorkDir": "/var/lib/docker/overlay2/26eff50ed5b9b32c0c3f03ab4e8d22700732d3b7a355fe976cdda0f998ddec3a/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/lib/modules",
                "Destination": "/lib/modules",
                "Mode": "ro",
                "RW": false,
                "Propagation": "rprivate"
            },
            {
                "Type": "volume",
                "Name": "a0f993c6084741f7a13078393a28464acb17f54caf5753f12fe538ae1a857400",
                "Source": "/var/lib/docker/volumes/a0f993c6084741f7a13078393a28464acb17f54caf5753f12fe538ae1a857400/_data",
                "Destination": "/var",
                "Driver": "local",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            }
        ],
        "Config": {
            "Hostname": "kind-worker2",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": true,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "container=docker"
            ],
            "Cmd": null,
            "Image": "kindest/node:latest",
            "Volumes": {
                "/var": {}
            },
            "WorkingDir": "",
            "Entrypoint": [
                "/usr/local/bin/entrypoint",
                "/sbin/init"
            ],
            "OnBuild": null,
            "Labels": {
                "io.x-k8s.kind.cluster": "kind",
                "io.x-k8s.kind.role": "worker"
            },
            "StopSignal": "SIGRTMIN+3"
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "c9b42b4d2733b2ab54ee02584cd4beeb236e26b4fdcc66f5852c3dca63943e09",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/c9b42b4d2733",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "kind": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "489eff3d5f02",
                        "kind-worker2"
                    ],
                    "NetworkID": "f525b267b89e6448f5e44d452b70dffab2dd3c1e840096268137336464db43ea",
                    "EndpointID": "028126fb87151f4dd4edf6bcd3a0b881ef58f503a5a3368af45d6f7f86da4442",
                    "Gateway": "172.18.0.1",
                    "IPAddress": "172.18.0.4",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "fc00:f853:ccd:e793::1",
                    "GlobalIPv6Address": "fc00:f853:ccd:e793::4",
                    "GlobalIPv6PrefixLen": 64,
                    "MacAddress": "02:42:ac:12:00:04",
                    "DriverOpts": null
                }
            }
        }
    }
]
```

**Comment by AkihiroSuda** [comment]
This is not implemented (yet) for the default `dockercompat` mode, but you can inspect the raw OCI config including mount information using `nerdctl container inspect --mode=native CONTAINER`.

**Comment by alegrey91** [comment]
Hi @AkihiroSuda, is there some specific reason why this is not yet implemented? I mean, this needs time or there's some blocking features?

**Comment by AkihiroSuda** [comment]
Implemented in #822
